Added coloring functionality

SitaGanesh opened the pull request:
# Description

Fill Mode: Adds a fill tool for coloring specific areas. Choose a color and click color button, and click on the desired area and BOOM!, if you what to add same color on different area, just click once again the color button and proceed. 

Fixed #458 

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added screenshots(if any) that prove my fix is effective or that my feature works
![image3](https://github.com/user-attachments/assets/c61e7538-7471-4888-b01c-211389d92bb1)

SitaGanesh committed:
Added coloring functionality

SitaGanesh committed:
Updated readme.md file

Documented the changes made in the project into readme.md file about adding the functionality details of color the drawing.

bislara reviewed (commented):
Well done, finally u did it. good job. 
one suggestion, if we are selecting color then keep it selected till we click back on pencil. 
I meant the color mode should not change to pencil on just one click.

SitaGanesh committed:
Updated script.js

 The change made is that if we are selecting color then keep it selected till we click back on pencil.
Nothing but  the color mode should not change to pencil on just one click

bislara commented:
Perfect. well done.
